server/modules/allocations.js

```diff
@@ -39,29 +43,26 @@ export const findPartnerById = partnerId => new Promise((resolve, reject) => {
     resolve(JSON.parse(result).values.filter(partner => partner.id === partnerId)[0]);
   });
 });
+/**
+ *@desc Fetches placements from allocations, based on the status specified
+ *
+ * @param {string} status - The status of placements to fetch
+ *
+ * @returns {axios} - Promise of the returned placements from allocations
+ */
+const fetchPlacementsByStatus = status => axios.get(`api/v1/placements?status=${status}`);
 
-
-// fetches placements based on status you passed
-export const fetchPlacementsByStatus = (status, callback) => {
-  axios.get(`api/v1/placements?status=${status}`)
-    .then((response) => {
-      callback(null, response.data.values);
-    })
-    .catch(error => callback(error, []))
-    .finally(() => client.quit());
-};
-
-export const fetchNewKickoffAllocation = (callback) => {
-  fetchPlacementsByStatus(placementStatus.onboarding, (err, response) => {
-    if (err) {
-      callback(err, []);
-      return;
-    }
-    const currentDate = new Date(Date.now());
-    const fromDate = currentDate.setDate(
-      currentDate.getDate() - Number(process.env.NUMBER_OF_DAYS),
-    ); // Specify desired date relative to present day
-    const newKickofEngagements = response.filter(data => (Date.parse(data.updated_at) >= fromDate));
-    callback(null, newKickofEngagements);
-  });
-};
+/**
+ * @desc Fetches new placements from the previous numberOfDays specified
+ *
+ * @param {string} status - The status of placements to fetch from allocations
+ * @param {number} numberOfDays - The range of days from which to get new placements
+ *
+ * @returns {array} - List of placements of specified status from the last numberOfDays
+ */
+export const fetchNewPlacements = (status, numberOfDays = 1) => fetchPlacementsByStatus(status).then((res) => {
+  const placements = res.data.values;
+  const currentDate = new Date(Date.now());
+  const fromDate = currentDate.setDate(currentDate.getDate() - numberOfDays);
+  return placements.filter(data => Date.parse(data.created_at) >= fromDate);
+});
```

Comment:
What if there is an error with the API-transaction to fetch the placements???

Reply:
That would be caught in the .catch() section by the programmer using this function

Reply:
Please provide an explanation about that in the JSDoc comments on this function.

Reply:
Done